app/assets/javascripts/discourse/app/lib/user-tips.js

```diff
@@ -3,111 +3,179 @@ import { iconHTML } from "discourse-common/lib/icon-library";
 import I18n from "I18n";
 import { escape } from "pretty-text/sanitizer";
 import tippy from "tippy.js";
+import isElementInViewport from "discourse/lib/is-element-in-viewport";
 
-const instances = {};
-const queue = [];
+const TIPPY_DELAY = 500;
+
+const instancesMap = {};
```

Comment:
`hideAllUserTips` does that and is in fact used in some tests.

Reply:
So my comment is mostly that the service would hold the state and not a global object window. I don't have the time to go a good demo of this but this would be a more "ember" way.